src/azure_cosmosdb/cosmosdb_endpoint.bal

```diff
@@ -0,0 +1,791 @@
+import ballerina/http;
+
+# Azure Cosmos DB Client object.
+public  client class Client {
+    private string baseUrl;
+    private string keyOrResourceToken;
+    private string host;
+    private string keyType;
+    private string tokenVersion;
+    private http:Client azureCosmosClient;
+
+    function init(AzureCosmosConfiguration azureConfig) {
+        self.baseUrl = azureConfig.baseUrl;
+        self.keyOrResourceToken = azureConfig.keyOrResourceToken;
+        self.host = azureConfig.host;
+        self.keyType = azureConfig.tokenType;
+        self.tokenVersion = azureConfig.tokenVersion;
+        http:ClientConfiguration httpClientConfig = {secureSocket: azureConfig.secureSocketConfig};
+        self.azureCosmosClient = new (self.baseUrl,httpClientConfig);
+    }
+
+    # Create a database inside a resource
+    # + databaseId -  id/name for the database
+    # + throughputProperties - Optional throughput parameter which will set 'x-ms-offer-throughput' or 
+    # 'x-ms-cosmos-offer-autopilot-settings' headers 
+    # + return - If successful, returns Database. Else returns error.  
+    public remote function createDatabase(string databaseId, ThroughputProperties? throughputProperties = ()) returns 
+    @tainted Database|error {
+        if self.keyType == TOKEN_TYPE_RESOURCE {
+            return prepareError("Enter a valid master key and token type should be master key");
+        }
+        json jsonPayload;
+        http:Request request = new;
+        string requestPath =  prepareUrl([RESOURCE_PATH_DATABASES]);
+        HeaderParameters header = mapParametersToHeaderType(POST, requestPath);
+        json body = {
+            id:databaseId
+        };
+        request.setJsonPayload(body);
+        request = check setHeaders(request, self.host, self.keyOrResourceToken, self.keyType, self.tokenVersion, header);
+        request = check setThroughputOrAutopilotHeader(request, throughputProperties);
+        var response = self.azureCosmosClient->post(requestPath, request);
+        [json, Headers] jsonreponse = check mapResponseToTuple(response);
+        return mapJsonToDatabaseType(jsonreponse);   
+    }
+
+    # Create a database inside a resource
+    # + databaseId -  id/name for the database
+    # + throughputProperties - Optional throughput parameter which will set 'x-ms-offer-throughput' or 
+    # 'x-ms-cosmos-offer-autopilot-settings' headers
+    # + return - If successful, returns Database. Else returns error.  
+    public remote function createDatabaseIfNotExist(string databaseId, ThroughputProperties? throughputProperties = ()) 
+    returns @tainted Database?|error {
+        if self.keyType == TOKEN_TYPE_RESOURCE {
+            return prepareError("Enter a valid master key and token type should be master key");
+        }
+        var result = self->getDatabase(databaseId);
+        if result is error {
+            return self->createDatabase(databaseId, throughputProperties);
+        }
```

Comment:
Here, error can happen for many purposes. 
We have to call the createDatabase only for the error of database already exists case.
Is it possible to validate the error properly?